### ISSUE TITLE
Implement pull of container image

### ISSUE THREAD

**Issue by stefwalter** [cockpit-project/cockpit#373]
Implement pull of an image. This happens when you click in the search dialog on an item that was searched for.

![docker-image-pull](https://cloud.githubusercontent.com/assets/795070/2547256/6a2b3bc8-b64d-11e3-984e-19a3c7c27c32.png)
- FWIW, I think the model of having to go through the search dialog is the right one. It lets us display whether an image is official/trusted/other before starting the pull.
- Implement DockerClient.pull('image').progres(function(fraction, message) { }).done(function() { })
  - This should call `waiting()/not_waiting()` logic for each image id that comes in, as it's coming in. So if the user clicks on a downloading image we see the right thing.
- We should add a DockerClient.progress = { } object in which each image being downloaded has a progress fraction.
- In render_image() if an image is in DockerClient.progress = { } then show the disk .bar-row as a progress instead just VirtualSize.
  - Use x/y instead of just x for value attribute.
- Pull API is here: http://docs.docker.io/en/latest/reference/api/docker_remote_api_v1.8/#create-an-image
  - You can get a feel for how pull API works with:
    `# printf "POST /images/create?fromImage=mvollmer/nodejs HTTP/1.0\r\nContent-Length: 0\r\n\r\n" | nc -U /var/run/docker.sock`
  - Response contains multiple json objects, ie: a stream.

Implementing cancellation of pull request is another task.
